### PR TITLE
Leverage m2e to run maven goal

### PR DESCRIPTION
**Pull request by jdneo**
Leverage m2e to run maven goal.

Issue: #967

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -6,6 +6,8 @@ Bundle-Version: 3.0.7.qualifier
 Bundle-Activator: com.microsoft.azuretools.core.Activator
 Require-Bundle: org.eclipse.ui,
  org.eclipse.core.runtime,
+ org.eclipse.jdt.launching,
```
Comment (Eskibear) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/META-INF/MANIFEST.MF]: is jdt.launching needed?

Comment (jdneo) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/META-INF/MANIFEST.MF]: Is it is.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/handlers/PublishHandler.java [hunk 2]
```diff
@@ -108,7 +113,13 @@ private void buildImage(IProject project) throws Exception {
         String destinationPath = null;
         if (MavenUtils.isMavenProject(project)) {
             destinationPath = MavenUtils.getTargetPath(project);
-            MavenUtils.executePackageGoal(project);
+            MavenExecuteAction action = new MavenExecuteAction(MAVEN_GOALS);
+            IContainer container;
+            container = MavenUtils.getPomFile(project).getParent();
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/handlers/PublishHandler.java]: same as above

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/handlers/PublishHandler.java [hunk 3]
```diff
@@ -108,7 +113,13 @@ private void buildImage(IProject project) throws Exception {
...
+            action.launch(container, MODE, () -> {
+                // TODO: callback after mvn package done. IMPORTANT
+                return null;
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/handlers/PublishHandler.java]: callback did nothing?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.webapp/src/com/microsoft/azuretools/webapp/ui/WebAppDeployDialog.java [hunk 4]
```diff
@@ -534,9 +534,7 @@ protected IStatus run(IProgressMonitor monitor) {
                 monitor.beginTask(message, IProgressMonitor.UNKNOWN);
                 try {
                     AzureDeploymentProgressNotification.notifyProgress(this, deploymentName, sitePath, 5, message);
-                    if (MavenUtils.isMavenProject(project)) {
-                        MavenUtils.executePackageGoal(project);
-                    } else {
+                    if (!MavenUtils.isMavenProject(project)) {
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.webapp/src/com/microsoft/azuretools/webapp/ui/WebAppDeployDialog.java]: why not put this part into the handler? which makes consistency that the package is ready before we show the dialog?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/handlers/DockerRunHandler.java [hunk 5]
```diff
@@ -103,11 +108,16 @@ public Object onExecute(ExecutionEvent event) throws ExecutionException {
             // export WAR file
             ConsoleLogger.info(String.format(Constant.MESSAGE_EXPORTING_PROJECT, destinationPath));
             if (MavenUtils.isMavenProject(project)) {
-                MavenUtils.executePackageGoal(project);
+                MavenExecuteAction action = new MavenExecuteAction(MAVEN_GOALS);
+                IContainer container;
+                container = MavenUtils.getPomFile(project).getParent();
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/handlers/DockerRunHandler.java]: container will always be got?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/handlers/PublishHandler.java (hunk 3 above)
Comment (Eskibear) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/handlers/PublishHandler.java]: will fix

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/handlers/DockerRunHandler.java (hunk 5 above)
Comment (Eskibear) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/handlers/DockerRunHandler.java]: in try/catch block, will print error and send telemetry

**Pull request merged**